db/hints/manager.cc

```diff
@@ -1364,6 +1364,19 @@ class directory_initializer::impl {
             dir_set.add_sharded(_hints_directory);
             manager_logger.debug("Creating and validating hint directories: {}", _hints_directory);
             co_await _dirs.create_and_verify(std::move(dir_set));
+            co_await lister::scan_dir(_hints_directory, {}, [this] (fs::path path, directory_entry de) -> future<> {
+                if (!de.type) {
+                    throw std::runtime_error(fmt::format("Unexpected entry {} with uknown type, in hints directory {}", path.native(), _hints_directory));
+                }
+                switch (*de.type) {
+                    case directory_entry_type::directory:
+                        co_await lister::rmdir(path);
+                    case directory_entry_type::regular:
+                        co_await remove_file(path.native());
+                    default:
+                        throw std::runtime_error(fmt::format("Unexpected entry {} with unexpected type, in hints directory {}", path.native(), _hints_directory));
+                }
+            });
```

Comment:
The file type verification is an overkill.
We already verify that in `create_and_verify`.
With that in mind, you can simply start with `co_await recursive_remove_directory(_hints_directory)`
before the call to `_dirs.create_and_verify`

Reply:
maybe under `if (co_await file_exists(_hints_directory))`
I think `recursive_remove_directory` returns an error if the directory doesn't exist.

Reply:
I considered that, but what if hints directory is in such a location that `scylla-server` has write access to it, but not to its parent dir? Is that even a plausible scenario?

Reply:
`create_and_verify` must be able to create it the first time so this can't happen